docs/helpers.rst

```diff
@@ -191,6 +191,26 @@ transaction support.  This is only required for fixtures which need
 database access themselves.  A test function would normally use the
 :py:func:`~pytest.mark.django_db` mark to signal it needs the database.
 
+``shared_db_wrapper``
+~~~~~~~~~~~~~~~~~~~~~
+
+This fixture can be used to create long-lived state in the database.
+It's meant to be used from fixtures with scope bigger than ``function``.
+It provides a context manager that will create a new database savepoint for you,
+and will take care to revert it when your fixture gets cleaned up.
+
+At the moment it does not work with ``transactional_db``,
+as the fixture itself depends on transactions.
+It also needs Django >= 1.6, as it's based on the ``atomic`` API.
+
+Example usage::
+
+  @pytest.fixture(scope='module')
+  def some_users(request, shared_db_wrapper):
+      with shared_db_wrapper(request):
```

Comment:
I can't use `with atomic():` because I have to register a callback on request :-/.

Thank you for pointing out transaction.set_rollback(True) - this will make my code cleaner (no more dummy exceptions!).

For the `using=...` part I might even be able to use Django's test case enter/exit atomics, or at least get some inspiration from there.